nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/resources/docs/org.apache.nifi.processors.splunk.QuerySplunkIndexingStatus/additionalDetails.html

```diff
@@ -72,5 +83,195 @@ <h3>Performance</h3>
     batch but the processor might execute the query with less number of undetermined events.
 </p>
 
+<h2>Example</h2>
+
+The following table represents four flow files being batch processed by the processor. For simplicity reasons "splunk.responded.at" time
+will be provided in HH:MM:SS format instead of Epoch time. The "Maximum waiting time" (referred as TTL from now on) is set for 3 minutes
+and the processor is scheduled to run every minute. The files are penalized for 30 seconds. "Undetermined" relationship is loop backed to the processor.
+
+<table>
+    <tbody>
+    <tr>
+        <th>Flow file id</th>
+        <th>Splunk acknowledgement id</th>
+        <th>Splunk responded at</th>
+        <th>Ack checked at Splunk</th>
+        <th>Relationship</th>
+    </tr>
+    <tr>
+        <td>flow file 1</td>
+        <td>null</td>
+        <td>10:46:02</td>
+        <td>No value set</td>
+    </tr>
+    <tr>
+        <td>flow file 2</td>
+        <td>0</td>
+        <td>10:46:18</td>
+        <td>No value set</td>
+    </tr>
+    <tr>
+        <td>flow file 3</td>
+        <td>1</td>
+        <td>10:47:04</td>
+        <td>No value set</td>
+    </tr>
+    <tr>
+        <td>flow file 4</td>
+        <td>2</td>
+        <td>10:47:23</td>
+        <td>No value set</td>
+    </tr>
+    </tbody>
+</table>
+
+<h4>First iteration</h4>
+
+The current time is 10:48:05 so all of our files are within TTL. Flow file 1 does not have a correct "splunk.acknowledgment.id"
+so will be directed to the "Failure" relationship. The remaining files will be sent for Splunk ack check.
+
+With the following Splunk response "acks:{[0:true, 1:false, 2:false]}" flow file 2 will be directed to "Success" relationship
+and the rest of the files to the "Undetermined" relationship and penalized for 30 seconds.
+
+<table>
+    <tbody>
+    <tr>
+        <th>Flow file id</th>
+        <th>Splunk acknowledgement id</th>
+        <th>Splunk responded at</th>
+        <th>Ack checked at Splunk</th>
+        <th>Relationship</th>
+    </tr>
+    <tr>
+        <td>flow file 1</td>
+        <td>null</td>
+        <td>10:46:02</td>
+        <td>No value set</td>
+        <td>Failure</td>
+    </tr>
+    <tr>
+        <td>flow file 2</td>
+        <td>0</td>
+        <td>10:46:18</td>
+        <td>true</td>
+        <td>Success</td>
+
+    </tr>
+    <tr>
+        <td>flow file 3</td>
+        <td>1</td>
+        <td>10:47:04</td>
+        <td>true</td>
+        <td>Undetermined</td>
+    </tr>
+    <tr>
+        <td>flow file 4</td>
+        <td>2</td>
+        <td>10:47:23</td>
+        <td>true</td>
+        <td>Undetermined</td>
+    </tr>
+    </tbody>
+</table>
+
+<h4>Second iteration</h4>
+
+The current time is 10:49:18 so both flow file 3 and flow file 4 are within TTL and they will be sent for Splunk ack check.
```

Comment:
As your example is pretty exhaustive, it might worth mentioning that during the timeframe of penalise, the processor might be triggered. In the case no further flow files arrived, it will abruptly finishes it's activation, in case there are new FFs are arrived, those will be processed (in this case: before the penalised ones)